examples/rl/sb3_simple.py

```diff
@@ -0,0 +1,170 @@
+""" Optuna example that optimizes the hyperparameters of
+a reinforcement learning agent using A2C implementation from Stable-Baselines3
+on a OpenAI Gym environment.
+
+This is a simplified version of what can be found in https://github.com/DLR-RM/rl-baselines3-zoo.
+
+You can run this example as follows:
+    $ python sb3_simple.py
+
+"""
+from typing import Any
+from typing import Dict
+
+import gym
+from stable_baselines3 import A2C
+from stable_baselines3.common.callbacks import EvalCallback
+import torch
+import torch.nn as nn
+
+import optuna
+from optuna.pruners import MedianPruner
+from optuna.samplers import TPESampler
+
+
+N_TRIALS = 100
+N_JOBS = 2
+N_STARTUP_TRIALS = 5
+N_EVALUATIONS = 2
+N_TIMESTEPS = int(2e4)
+EVAL_FREQ = int(N_TIMESTEPS / N_EVALUATIONS)
+N_EVAL_EPISODES = 3
+
+ENV_ID = "CartPole-v1"
+
+DEFAULT_HYPERPARAMS = {
+    "policy": "MlpPolicy",
+    "env": ENV_ID,
+}
+
+
+def sample_a2c_params(trial: optuna.Trial) -> Dict[str, Any]:
+    """Sampler for A2C hyperparameters."""
+    gamma = trial.suggest_categorical("gamma", [0.9, 0.95, 0.98, 0.99, 0.999, 0.9999])
+    max_grad_norm = trial.suggest_categorical("max_grad_norm", [0.3, 0.5, 0.7, 0.9, 1, 2, 5])
+    gae_lambda = trial.suggest_categorical("gae_lambda", [0.8, 0.9, 0.92, 0.95, 0.98, 0.99, 1.0])
+    n_steps = trial.suggest_categorical("n_steps", [8, 16, 32, 64, 128, 256, 512, 1024])
```

Comment:
`suggest_categorical` does not consider the value of each choice, so the performance might degrade for real-valued parameters. On the other hand, these `suggest_categorical` calls can be converted to `suggest_float` or `suggest_int` as follows.

```suggestion
    gamma = 1. - trial.suggest_float("gamma", 0.0001, 0.1, log=True)
    max_grad_norm = trial.suggest_float("max_grad_norm", 0.3, 5.0, log=True)
    gae_lambda = 1. - trial.suggest_float("gae_lambda", 0.001, 0.2. log=True)
    n_steps = 2 ** trial.suggest_int("exponent_n_steps", 3, 10)
```

Reply:
thanks, I did not know about those...

Reply:
btw, why do you set `log=True` ?